### PR TITLE
[Movement] Update stage lifting/lowering

### PR DESCRIPTION
**Pull request by maltewae**
Updates the z lift and lifting behaviour of stages during automatic movement of stages to device.

The following is need:
- Updated Path Finding Algo: Now you can set a z level in chip coordinates for each stage. If given the the stage, will also keep the z coordinate at this level. Note that we keep the chip coordinate z level equal, not the stage coordinate z level. This coordinate will be transformed in stage space. This can incorporate a slight tilt of the chip. If no level is z, the algo always tries to keep the z coordinate the same as before.
- Enhanced the lifting logic: Now we keep track of lifted stages individually. The idea is that we only lift or lower stages, if the stage was lowered or lifted before.

### CODE REVIEW

File: LabExT/Movement/MoverNew.py [hunk 1]
```diff
@@ -602,17 +615,40 @@ def lift_stages(self):
...
+                    target_coordinate = ChipCoordinate(
+                        x=stage_start_pos.x, y=stage_start_pos.y, z=target_z)
+                    try:
+                        calibration.move_absolute(
+                            target_coordinate, wait_for_stopping=True)
+                    finally:
+                        stage_end_pos = calibration.get_position()
+                        lift_delta = stage_end_pos - stage_start_pos
+
+                        if not np.isclose(
```
Comment (maltewae) [inline LabExT/Movement/MoverNew.py:637]: @marcoep @boris-vukovic Need same feedback on the tolerances.
```
absolute(a - b) <= (atol + rtol * absolute(b))
```
`atol` and `rtol` are the default ones at the moment.

File: LabExT/Movement/MoverNew.py [hunk 2]
```diff
@@ -602,17 +615,44 @@ def lift_stages(self):
...
```
Comment (marcoep) [inline LabExT/Movement/MoverNew.py:637]: Why are you doing this check here? Why could this fail?

File: LabExT/Movement/MoverNew.py [hunk 3]
```diff
@@ -517,7 +525,12 @@ def move_absolute(
                 raise MoverError(
                     f"No {orientation} stage configured, but target coordinate for {orientation} passed.")
 
-            path_planning.set_stage_target(calibration, target)
+            current_stage_level = None
+            if calibration in self._lifted_stages:
+                current_stage_level = self.z_lift
```
Comment (marcoep) [inline LabExT/Movement/MoverNew.py:530]: At this point your stages should be lifted anyhow.

I cannot see a reason why you would pass a z_level to the path planning. Doesn't path planning just move on a fixed z-level? If you start lifted, the path planning should stay on the lifted z-level anyhow, no?

File: LabExT/Movement/MoverNew.py [hunk 4]
```diff
@@ -602,17 +615,40 @@ def lift_stages(self):
...
         """
 
-        def _lift_lower_stages(lift):
+        def _toggle_stage_lift():
+            """
+            Moves all stages to z lift level.
+            Moves up if stages_are_lifted is false and vice versa.
+            """
             for calibration in self.calibrations.values():
+                _is_lifted = calibration in self._lifted_stages
```
[the remark this answers is not in the document]
Comment (marcoep) [reply LabExT/Movement/MoverNew.py:624]: Why do you error check the lifting movement? Is this prone to fail?

**Comment by maltewae** [comment]
See #103

**Pull request closed without merging**